packages/snaps-sdk/src/types/permissions.ts

```diff
@@ -73,3 +73,72 @@ export type InitialPermissions = Partial<{
   wallet_snap: Record<string, RequestedSnap>;
   /* eslint-enable @typescript-eslint/naming-convention */
 }>;
+
+export type CaveatConstraint = {
+  /**
+   * The type of the caveat. The type is presumed to be meaningful in the
+   * context of the capability it is associated with.
+   *
+   * In MetaMask, every permission can only have one caveat of each type.
+   */
+  readonly type: string;
+
+  /**
+   * Any additional data necessary to enforce the caveat.
+   */
+  readonly value: Json;
+};
+
+/**
+ * A `ZCAP-LD`-like permission object. A permission is associated with a
+ * particular `invoker`, which is the holder of the permission. Possessing the
+ * permission grants access to a particular restricted resource, identified by
+ * the `parentCapability`. The use of the restricted resource may be further
+ * restricted by any `caveats` associated with the permission.
+ *
+ * Note: this type is mirroring: https://github.com/MetaMask/core/blob/main/packages/permission-controller/src/Permission.ts#L39
+ */
+export type PermissionConstraint = {
+  /**
+   * The context(s) in which this capability is meaningful.
+   *
+   * It is required by the standard, but we make it optional since there is only
+   * one context in our usage (i.e. the user's MetaMask instance).
+   */
+  readonly '@context'?: NonEmptyArray<string>;
+
+  /**
+   * The caveats of the permission.
+   *
+   * @see {@link Caveat} For more information.
+   */
+  readonly caveats: null | NonEmptyArray<CaveatConstraint>;
+
+  /**
+   * The creation date of the permission, in UNIX epoch time.
+   */
+  readonly date: number;
+
+  /**
+   * The GUID of the permission object.
+   */
+  readonly id: string;
+
+  /**
+   * The origin string of the subject that has the permission.
+   */
+  readonly invoker: string;
+
+  /**
+   * A pointer to the resource that possession of the capability grants
+   * access to, for example a JSON-RPC method or endowment.
+   */
+  readonly parentCapability: string;
+};
+
+/**
+ * Permissions associated with a PermissionController subject.
+ * Note: this is mirroring https://github.com/MetaMask/core/blob/main/packages/permission-controller/src/PermissionController.ts#L138
```

Comment:
Can you add this note at the top to mention that these are copied over? Instead of using a dependency for bundle sizing reasons

Reply:
Added.